lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/Builtin.java

```diff
@@ -24,28 +120,303 @@
     String description() default "";
 
     /**
-     * @return When applied to a method/constructor with varargs, indicates how many times vararg
-     *     parameter should be expanded and implicitly how many builtin nodes should be generated.
-     *     Must be zero when there are no varaargs in the parameters list.
+     * Element indicates how many times a method with vararg parameter should be expanded and
+     * implicitly how many builtin nodes should be generated. Must be zero when there are no
+     * varaargs in the parameters list.
+     *
+     * <pre>
+     * class Foo {
+     *    {@link Builtin.Method @Builtin.Method}(expandVarargs = 2)
+     *     public static Foo create(Object items...) {
+     *         // ...
+     *     }
+     * }
+     * </pre>
+     *
+     * will generate 2 classes
+     *
+     * <pre>
+     * {@link BuiltinMethod @BuiltinMethod}(type = "Foo", name = "create_1")
+     * public class Create1FooNode extends Node {
+     *   Foo execute(Foo _this, Object items_1) {
+     *     return Foo.create(items_1);
+     *   }
+     * }
+     * {@link BuiltinMethod @BuiltinMethod}(type = "Foo", name = "create_2")
+     * public class Create2FooNode extends Node {
+     *   Foo execute(Foo _this, Object items_1, Object items_2) {
+     *     return Foo.create(items_1, items_2);
+     *   }
+     * }
+     * </pre>
+     *
+     * @return number of desired expansions for vararg parameter.
      */
     int expandVarargs() default 0;
   }
 
   /**
-   * Annotation indicating that a potential exception during the execution of the method should be
-   * wrapped in Enso's error type that will be reported to the user. The annotation can be repeated
-   * leading to multiple catch clauses.
+   * Annotation indicating that compile- or run-time exception should be wrapped in Enso's error
+   * type that will be reported to the user. The annotation can be repeated leading to multiple
+   * catch-clauses.
+   *
+   * <p>Consider simple method with possible runtime exception:
+   *
+   * <pre>
+   * class Foo {
+   *   private Object[] items;
+   *   {@link Builtin.Method @Builtin.Method}
+   *   {@link Builtin.WrapException @Builtin.WrapException}(from=IndexOutOfBounds.class, to=InvalidArrayIndexError.class)
+   *   public Object get(long index) {
+   *     return items[index];
+   *   }
+   * </pre>
+   *
+   * Given the information from {@link WrapException#from()} and {@link WrapException#to()}
+   * elements, the processor knows that the call to {@code get()} has to be wrapped in try/catch,
+   * catching and re-throwing values of a particular type, respectively. The processor will
+   * automatically infer what parameters need to be passed to the new exception using the
+   * information from the {@link WrapException#to()} class.
```

Comment:
I'm not sure if I understand how is it inferred what should be passed to the constructor of the target exception type.

Reply:
Added a few comments. Hopefully it is a bit more clear now